### PR TITLE
Modified code to add the flags to the method declaration

### PR DESCRIPTION
**Pull request by Fare9**

### CODE REVIEW

File: shuriken/mjolnir/include/mjolnir/MjolnIRDialect.td [hunk 1]
```diff
@@ -89,5 +89,35 @@ def MjolnIR_FieldTypeAttr : I32EnumAttr<
...
+       I32BitEnumAttrCase<"NATIVE", 0x100, "native">,
+       I32BitEnumAttrCase<"INTERFACE", 0x200, "interface">,
+       I32BitEnumAttrCase<"ABSTRACT", 0x400, "abstract">,
+       I32BitEnumAttrCase<"STRICT", 0x800, "strict">,
+       I32BitEnumAttrCase<"SYNTHETIC", 0x1000, "synthetic">,
+       I32BitEnumAttrCase<"ANNOTATION", 0x2000, "annotation">,
+       I32BitEnumAttrCase<"ENUM", 0x4000, "enum">,
+       I32BitEnumAttrCase<"UNUSED", 0x8000, "unused">,
+       I32BitEnumAttrCase<"CONSTRUCTOR", 0x10000, "constructor">,
+       I32BitEnumAttrCase<"DECLARED_SYNCHRONIZED", 0x20000, "declared_synchronized">
```
Comment (badumbatish) [inline shuriken/mjolnir/include/mjolnir/MjolnIRDialect.td:117]: I'm sure its nothing but why is it 0x8 for STATIC and then going to 0x10 for FINAL and then multiplying by 2 again. Is this a conventional thing in smali/dex?

File: shuriken/mjolnir/include/mjolnir/MjolnIROps.td [hunk 2]
```diff
@@ -295,46 +295,85 @@ def MethodOp : MjolnIR_Op<"method", [
...
+        bool isStatic() { return hasFlag(MethodFlags::STATIC); }
+
+        /// Check if method is final
+        bool isFinal() { return hasFlag(MethodFlags::FINAL); }
+
+        /// Check if method is synchronized
+        bool isSynchronized() { return hasFlag(MethodFlags::SYNCHRONIZED); }
+
+        /// Check if method is a constructor
+        bool isConstructor() { return hasFlag(MethodFlags::CONSTRUCTOR); }
```
Comment (badumbatish) [inline shuriken/mjolnir/include/mjolnir/MjolnIROps.td:372]: nitpicking but mergable, but this should add a `const` behind function name maybe?

**Review by badumbatish — APPROVED** [review]

File: shuriken/mjolnir/include/mjolnir/MjolnIRDialect.td (hunk 1 above)
Comment (Fare9) [reply shuriken/mjolnir/include/mjolnir/MjolnIRDialect.td:117]: Yes, this is directly obtained from Smali documentation.

File: shuriken/mjolnir/include/mjolnir/MjolnIROps.td (hunk 2 above)
Comment (Fare9) [reply shuriken/mjolnir/include/mjolnir/MjolnIROps.td:372]: Yes, that's fine

**Pull request merged**